daemon/api.go

```diff
@@ -1672,6 +1675,66 @@ type createResponseData struct {
 	SSHKeyCount int `json:"ssh-key-count"`
 }
 
+var userLookup = user.Lookup
+
+func addAuthUser(st *state.State, username string) error {
+	// create a new (local) macaroon
+	rootKey := make([]byte, 32)
+	if _, err := rand.Read(rootKey); err != nil {
+		return fmt.Errorf("cannot get random bytes: %s", err)
+	}
+	// FIXME: is this a proper use of ID and Location?
+	maca, err := macaroon.New(rootKey, username, "local-user")
```

Comment:
ID and location are visible for anyone having acess to the macaroon, and are supposed to help identify the key to verify it.
I don't see you keeping the rootKey value around for a later macaroon verification. This key is used to sign the macaroon and it will be needed to perform the macaroon verification (ie. check its integrity when getting it back). For store macaroons that validation is done in the server, wondering if we should have local macaroons setup for every user and do the local verification on API requests, or we don't need to verify the macaroon and just check if we have it setup in the state?